### ISSUE TITLE
Extra whitespace inserted after " in smarty modifier, when applied to variable property of object

### ISSUE THREAD

**Issue by JordyVanderstukken** [aswinkumar863/smarty-vscode-support#292]
When you format this:
`<input type=text name='{$name}' value="{$item->{$name}|date_format:"%d-%m-%Y"}" />`

it turns into this:
`<input type=text name='{$name}' value="{$item->{$name}|date_format:" %d-%m-%Y"}" />`

Note the extra space at the start of the date format. This can cause issues when a specific valeu is expected, which now has an extra space appended to it. The issue does not occur with single quotes, only double, and only occurs if you apply a smarty modifier to a variable property. (I put a simple "$name" as the variable property name here for ease of debugging, but a more complex one can be used in which case this syntax is necessary.)

I'm not using any other extensions that could interfere with this behaviour.

**Comment by aswinkumar863** [comment]
@JordyVanderstukken 

You are right, this unexpected behaviour is due to the nested curly brace. In some cases, the nested braces can even go deeper like 

```php
print_r($key1->{$key2->{$key3->val}});
```
Handling nested curly braces while formatting brings additional problems such as performance and validating balanced pairs. So using alternative approaches/workaround would be the solution.


**Here are some other approaches I recommended**

```tpl
<input type=text name='{$name}' value='{$item->{$name}|date_format:"%d-%m-%Y"}' />
<input type=text name='{$name}' value="{$item->{$name}|date_format:'%d-%m-%Y'}" />
<input type=text name='{$name}' value="{$item->$name|date_format:'%d-%m-%Y'}" />
```
Using smarty variables or [PHP 7's interpretation](https://www.php.net/manual/en/migration70.incompatible.php#migration70.incompatible.variable-handling.indirect) are also can be used.

Thanks for reporting. Keep supporting.
